### ISSUE TITLE
Perlless activation fails with missing time zone file

### ISSUE THREAD

**Issue by awebeer256** [NixOS/nixpkgs#321280]
### Describe the bug
If I try to make my system perlless, my system rebuild fails.

### Steps To Reproduce
1. Add the following module to your system configuration: 
```
{ modulesPath, ... }:
{
  imports = [ "${modulesPath}/profiles/perlless.nix" ];
}
```
2. `nixos-rebuild`

### Expected behavior
The command succeeds.

### Screenshots

Commands:
```
$ doas nixos-rebuild boot
building the system configuration...
error: builder for '/nix/store/<hash>-etc-lowerdir.drv' failed with exit code 1;
       last 1 log lines:
       > cp: cannot stat '/etc/zoneinfo/<region>/<city>': No such file or directory
       For full logs, run 'nix log /nix/store/<hash>-etc-lowerdir.drv'.
error: 1 dependencies of derivation '/nix/store/<hash>-nixos-system-<hostname>-<timestamp>.<hash>.drv' failed to build

$ nix log /nix/store/<hash>-etc-lowerdir.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/<hash>-etc-lowerdir.drv^*'
cp: cannot stat '/etc/zoneinfo/<region>/<city>': No such file or directory

$ nix log '/nix/store/<hash>-etc-lowerdir.drv^*'
cp: cannot stat '/etc/zoneinfo/<region>/<city>': No such file or directory

$ stat /etc/zoneinfo/<region>/<city>
  File: /etc/zoneinfo/<region>/<city>
  Size: xxxx            Blocks: x          IO Block: xxxx   regular file
Device: x,xx    Inode: xxxxxxx     Links: 1
Access: (0444/-r--r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: <timestamp>
Modify: <timestamp>
Change: <timestamp>
 Birth: <timestamp>
```

### Additional context
`nixos-rebuild dry-build` succeeds.
I use flakes.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.4, NixOS, 24.05 (Uakari), 24.05.20240612.cc54fb4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.90.0-beta.1`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/aa0zsahvyqnvzkicsh29jirm9hwv95pg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
Odd, thought #314579 was meant to fix this. Perhaps more digging is needed.

**Comment by ivan770** [comment]
What is the revision of nixpkgs that you are trying to build your system with? #314579 is merged without a 24.05 backport.

**Comment by eclairevoyant** [comment]
Per nix-info they're on 24.05 not unstable, so that makes sense. Is it backportable?

**Comment by ivan770** [comment]
> Is it backportable?

The PR itself is backportable, but I'm not sure if it complies with nixpkgs' [backport criteria](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-acceptable-for-releases).

**Comment by eclairevoyant** [comment]
Right, that's what I meant to ask - is it a breaking change? or simply a bugfix with low chance of regressions?

**Comment by ivan770** [comment]
> Right, that's what I meant to ask - is it a breaking change? (i.e. is it possible for someone to rely on the old behaviour?) or simply a bugfix with low chance of regressions?

Is it not a breaking change because the `system.etc.overlay.enable` option is marked as experimental (and also because the behavior before the PR was erroneous in the first place).

**Comment by azahi** [comment]
Fixed in #321794.
